SU2_CFD/src/objectives/turbomachinery/CTurbomachineryPerformance.cpp

```diff
@@ -0,0 +1,318 @@
+                                                  unsigned short nDim,
+                                                  su2double areaIn,
+                                                  su2double radiusIn,
+                                                  su2double areaOut,
+                                                  su2double radiusOut) : CTurbomachineryBladePerformance(fluidModel, nDim, areaIn, radiusIn, areaOut, radiusOut){}
+
+void CTurbineBladePerformance::ComputePerformance(const CTurbomachineryCombinedPrimitiveStates &primitives) {
+
+  /*--- Compute Inlet and Outlet state ---*/
+  InletState.ComputeState(FluidModel, primitives.GetInletPrimitiveState());
+  OutletState.ComputeState(FluidModel, primitives.GetOutletPrimitiveState());
+
+  /*--- Compute isentropic Outflow quantities ---*/
+  FluidModel.SetTDState_Ps(OutletState.GetPressure(), InletState.GetEntropy());
+  auto enthalpyOutIs = FluidModel.GetStaticEnergy() + OutletState.GetPressure() / FluidModel.GetDensity();
+  auto tangVel = primitives.GetOutletPrimitiveState().GetTangVelocity();
+  auto relVelOutIs2 = 2 * (OutletState.GetRothalpy() - enthalpyOutIs) + tangVel * tangVel;
+
+  /*--- Compute performance ---*/
+  EntropyGen = (OutletState.GetEntropy() - InletState.GetEntropy()); // / abs(InletState.GetEntropy() + 1);
+  EulerianWork = InletState.GetTotalEnthalpy() - OutletState.GetTotalEnthalpy();
+  TotalPressureLoss = (InletState.GetTotalRelPressure() - OutletState.GetTotalRelPressure()) /
+                      (InletState.GetTotalRelPressure() - OutletState.GetPressure());
+  KineticEnergyLoss = 2 * (OutletState.GetEnthalpy() - enthalpyOutIs) / relVelOutIs2;
+}
+
+CCompressorBladePerformance::CCompressorBladePerformance(CFluidModel& fluidModel,
+                                                        unsigned short nDim,
+                                                        su2double areaIn,
+                                                        su2double radiusIn,
+                                                        su2double areaOut,
+                                                        su2double radiusOut) : CTurbomachineryBladePerformance(fluidModel, nDim, areaIn, radiusIn, areaOut, radiusOut){}
+
+void CCompressorBladePerformance::ComputePerformance(const CTurbomachineryCombinedPrimitiveStates &primitives) {
+  /*--- Compute Inlet and Outlet state ---*/
+  InletState.ComputeState(FluidModel, primitives.GetInletPrimitiveState());
+  OutletState.ComputeState(FluidModel, primitives.GetOutletPrimitiveState());
+
+  /*--- Compute isentropic Outflow quantities ---*/
+  FluidModel.SetTDState_Ps(OutletState.GetPressure(), InletState.GetEntropy());
+  auto enthalpyOutIs = FluidModel.GetStaticEnergy() + OutletState.GetPressure() / FluidModel.GetDensity();
+  auto tangVel = primitives.GetOutletPrimitiveState().GetTangVelocity();
+  auto relVelOutIs2 = 2 * (OutletState.GetRothalpy() - enthalpyOutIs) + tangVel * tangVel;
```

Comment:
Don't use auto for the return of computations with su2doubles, causes AD problems, use su2double explicitly.